backend/lib/richard_burton_web/controllers/publication_controller.ex

```diff
@@ -76,6 +76,18 @@ defmodule RichardBurtonWeb.PublicationController do
     conn |> put_status(status) |> json(Publication.Codec.flatten(response_body))
   end
 
+  def update(conn, %{"id" => id} = params) do
+    publication = Publication.get!(id) |> Publication.preload()
+    result = publication |> Publication.update(params)
```

Comment:
Would it be interesting to do it in a single pipeline?

```elixir
result = id
  |> Publication.get!()
  |> Publication.preload()
  |> Publication.update(params)
```

Also, should we catch the error thrown by `get!` and return a response explicitly? Otherwise, it will be a 500 when it could be a 404

Reply:
this error is handled by Phoenix.Endpoint plug and the response status code and body are correctly set unless the `debug_error` flag is on in the environment's config.